### PR TITLE
lwip: menuconfig option for max number of SNTP servers (IDFGH-5616)

### PR DESCRIPTION
**Pull request by vortigont**
LWIP has two define's for setting max number of sntp servers:
 - total number of handled servers
 - max number of sntp's picked via DHCP

by default both values are equal to 1, but could be set separately.
This is a followup for espressif/arduino-esp32#5343. Having this menuconfig option allows building lwip for arduino-esp32 with multiple sntps's supported

### CODE REVIEW

**Comment by Alvin1Zhang** [comment]
Thanks for your contribution.

File: components/lwip/port/esp32/include/lwipopts.h [hunk 1]
```diff
@@ -1074,6 +1074,15 @@ LWIP_FORWARD_DECLARE_C_CXX void sntp_sync_time(struct timeval *tv);
 
 LWIP_FORWARD_DECLARE_C_CXX uint32_t sntp_get_sync_interval(void);
 
+// Max number of SNTP servers handled (default equal to LWIP_DHCP_MAX_NTP_SERVERS)
+#if defined CONFIG_LWIP_SNTP_MAX_SERVERS && !defined SNTP_MAX_SERVERS
```
Comment (david-cermak) [inline components/lwip/port/esp32/include/lwipopts.h:1078]:
```suggestion
#if defined CONFIG_LWIP_SNTP_MAX_SERVERS
```
As we don't allow to override `CONFIG_...` macros defined in configuration menu.

File: components/lwip/Kconfig [hunk 2]
```diff
@@ -846,6 +846,23 @@ menu "LWIP"
...
+                SNTP callback for such replies should be set accordingly (see sntp_servermode_dhcp() func.)
+
+        config LWIP_DHCP_MAX_NTP_SERVERS
+            int "Maximum number of NTP servers aquired via DHCP"
+            default 1
+            range 1 16
+            depends on LWIP_DHCP_GET_NTP_SRV
+            help
+                Set maximum number of NTP servers aquired via DHCP-offer.
+                Must be less or equal to total "Maximum number of NTP servers".
```
Comment (david-cermak) [inline components/lwip/Kconfig:864]: Question: Does the lwip throws a compile `#error` if it `> SNTP_MAX_SERVERS` ? maybe we should if it doesn't.

**Review by david-cermak — APPROVED** [review]
LGTM, just a question and an optional suggestion.

File: components/lwip/port/esp32/include/lwipopts.h (hunk 1 above)
Comment (vortigont) [reply components/lwip/port/esp32/include/lwipopts.h:1078]: OK, I will amend the change into last commit to keep it clean.

File: components/lwip/Kconfig (hunk 2 above)
Comment (vortigont) [reply components/lwip/Kconfig:864]: No, it does not. Actually if I get this right any extra DHCP-servers would be just silently ignored. Not sure that throwing an error and stop here is a good idea.
@david-cermak I can make a test setup with dhcp-server and confirm this if you like?

**Comment by vortigont** [comment]
@david-cermak I've updated notes for menuoption, also added NTP  debug option and sntp example to utilize SNTP via DHCP if build-in.
Would appreciate if you could review changes.
Thanks!

**Comment by david-cermak** [comment]
@vortigont Thanks for the prompt update and addressing the concern of misconfiguration.
This PR looks very good to me and has been added to the internal round of testing and reviews (this usually takes some time though till it gets merged to master).

**Pull request merged**